multihash/codecs.py

```diff
@@ -48,7 +48,7 @@ class CodecReg(metaclass=_CodecRegMeta):
         ('base64', base64.b64encode, base64.b64decode)]
     if base58:
         _common_codec_data.append(
-            ('base58', lambda s: bytes(base58.b58encode(s)), base58.b58decode))
+            ('base58', lambda s: bytes(base58.b58encode(s), 'utf-8'), base58.b58decode))
```

Comment:
Since base58 outputs ASCII when encoding, I think the correct option would be to use ``'ascii'`` instead of ``'utf-8'``. Thus, if some non-ASCII code point erroneously slipped in, we would get a ``UnicodeDecodeError``.